### ISSUE TITLE
Enum element with name `false` not compiling

### ISSUE THREAD

**Issue by NoelAbrahams** [Microsoft/TypeScript#12963]
**TypeScript Version:**  2.1.4

**Code**

```ts
const enum Foo {
    bar,
    false
}

// Okay
function aa(): Foo.bar {

    return Foo.bar;
}

// Error: Identifier expected
function bb(): Foo.false {

    return Foo.false;
}
```

**Expected behavior:**

No error

**Actual behavior:**

Has error

**Comment by RyanCavanaugh** [comment]
This should be an easy fix if anyone's interested in sending a PR

**Comment by jkascend** [comment]
I'm new to the repo but I'd like to try and tackle it if that's alright.

I could maybe use some guidance on where to start though.

**Comment by RyanCavanaugh** [comment]
@jkascend in `src/compiler/parser.ts` there's `parseIdentifier` and `parseIdentifierName` and we're calling the wrong one (with some level of indirection) from `parseEnumDeclaration`

**Comment by jkascend** [comment]
@RyanCavanaugh thanks, glad I was at least looking in the right file before asking for help. :)

So, based on your direction, I've traced the code governing which of those functions (`parseIdentifier` or `parseIdentifierName`) is called for "Foo.false" in OP's example code to here: https://github.com/Microsoft/TypeScript/blob/master/src/compiler/parser.ts#L2045

When parsing the right side of the dot it's passing `false` for `allowReserveWords`. If I set this to `true`, we call the "correct" function, in as much as now that test code transpiles without giving an error.

I added a test with code like:
`const enum Foo {
    false
}

function f(): Foo.false {
    return Foo.false;
}`

and after making that change it outputs the below baseline JS, which seems correct to me:

`function f() {
    return 0 /* false */;
}`

Am I on the right track with this? Making this change did break the test named `tests/cases/conformance/parser/ecmascript5/parservoidInQualifiedName2.ts`, which tells me simply flipping that boolean is not an acceptable solution.

Here's the error output from that failed test:
`tests/cases/conformance/parser/ecmascript5/parservoidInQualifiedName2.ts(1,9): error TS2503: Cannot find namespace 'x'.


==== tests/cases/conformance/parser/ecmascript5/parservoidInQualifiedName2.ts (1 errors) ====
    var v : x.void;
            ~
!!! error TS2503: Cannot find namespace 'x'.`
